### PR TITLE
[TSK-202] update root and services README

### PR DESCRIPTION
**Pull request by Hoocki**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,44 @@
...
+## Goals
+- [x] Apply a monorepositional approach to the interaction between services.
+- [x] Design 7 services to achieve high reliability and scalability of the system.
+- [x] Use docker containers to run external components independently.
+- [x] Creating migration scripts for database version control.
+- [x] Configure the CI-pipeline to automatically build the project, run tests.
+- [x] Use Apache Kafka to store transaction data in order to increase fault tolerance.
+- [x] Create tests with code coverage for each service over 80%.
+- [ ] Implementation of authentication and authorization using JWT.
+- [ ] Implementation of API-Gateway and Service Discovery to simplify communication between services and clients.
```
Comment (adsukharev) [inline README.md:16]: I would add `create frontend app

File: README.md [hunk 2]
```diff
@@ -1,2 +1,44 @@
...
+The project is aimed at implementing a payment system with QR code generation in a microservice architecture.
+
+## Goals
+- [x] Apply a monorepositional approach to the interaction between services.
+- [x] Design 7 services to achieve high reliability and scalability of the system.
+- [x] Use docker containers to run external components independently.
+- [x] Creating migration scripts for database version control.
+- [x] Configure the CI-pipeline to automatically build the project, run tests.
+- [x] Use Apache Kafka to store transaction data in order to increase fault tolerance.
+- [x] Create tests with code coverage for each service over 80%.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- [x] Create unit and integration tests with code coverage over 80%.
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,44 @@
...
+
+## Description
+The project is aimed at implementing a payment system with QR code generation in a microservice architecture.
+
+## Goals
+- [x] Apply a monorepositional approach to the interaction between services.
+- [x] Design 7 services to achieve high reliability and scalability of the system.
+- [x] Use docker containers to run external components independently.
+- [x] Creating migration scripts for database version control.
+- [x] Configure the CI-pipeline to automatically build the project, run tests.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- [x] Configure the CI-pipeline for GitHub to automatically build the project, run tests.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,44 @@
 # QPay
-### Payment system with QR code
+### Quick payment system by generating a QR code.
+
+## Description
+The project is aimed at implementing a payment system with QR code generation in a microservice architecture.
+
+## Goals
+- [x] Apply a monorepositional approach to the interaction between services.
+- [x] Design 7 services to achieve high reliability and scalability of the system.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- [x] Design microservices to achieve high reliability and scalability of the system.
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,44 @@
...
+Architecture: https://drive.google.com/file/d/1Jn5VrUUoQQtLY0N9Jk_7VIYwRXwG2yOq/view?usp=sharing
+- **User manager** - stores and processes user information.
+- **Payment manager** - stores and processes information about wallets with further sending of data to other services
+  (Notification and Transaction manager).
+- **Notification manager** - receives data from ***Apache Kafka*** (data from the Payment manager is first sent to Kafka),
+  then sends a message to the user's email with information about the operations performed on the wallet.
+- **Transaction manager** - stores information about completed transactions over wallets.
+- **QR generator** - generates a QR-code based on transaction information: amount of money, sender, recipient.
+- **Auth manager** - performs user authentication and authorization in the system.
+- **Service Discovery** - responsible for establishing a connection between services and clients.
```
Comment (adsukharev) [inline README.md]: Add API-Gateway

File: README.md [hunk 6]
```diff
@@ -1,2 +1,44 @@
...
+- [x] Use docker containers to run external components independently.
+- [x] Creating migration scripts for database version control.
+- [x] Configure the CI-pipeline to automatically build the project, run tests.
+- [x] Use Apache Kafka to store transaction data in order to increase fault tolerance.
+- [x] Create tests with code coverage for each service over 80%.
+- [ ] Implementation of authentication and authorization using JWT.
+- [ ] Implementation of API-Gateway and Service Discovery to simplify communication between services and clients.
+
+## Structure
+Project contains 7 services,
```
Comment (adsukharev) [inline README.md]: 8
But maybe skip number, if later you add new services.
Just: Project contains the following services

File: README.md [hunk 7]
```diff
@@ -1,2 +1,44 @@
...
+each of which is responsible for performing key tasks to maintain the project's performance.
```
Comment (adsukharev) [inline README.md]: I would delete this sentence, it is just noise

File: README.md [hunk 8]
```diff
@@ -1,2 +1,44 @@
 # QPay
-### Payment system with QR code
+### Quick payment system by generating a QR code.
```
Comment (adsukharev) [inline README.md]:
```suggestion
### Payment system by generating QR codes.
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,44 @@
...
+  (Notification and Transaction manager).
+- **Notification manager** - receives data from ***Apache Kafka*** (data from the Payment manager is first sent to Kafka),
+  then sends a message to the user's email with information about the operations performed on the wallet.
+- **Transaction manager** - stores information about completed transactions over wallets.
+- **QR generator** - generates a QR-code based on transaction information: amount of money, sender, recipient.
+- **Auth manager** - performs user authentication and authorization in the system.
+- **Service Discovery** - responsible for establishing a connection between services and clients.
+
+## How to run
+- Clone this project to your computer.
```
Comment (adsukharev) [inline README.md]: remove it

File: README.md [hunk 10]
```diff
@@ -1,2 +1,44 @@
...
+- Download docker.
```
Comment (adsukharev) [inline README.md]: remove

File: README.md [hunk 11]
```diff
@@ -1,2 +1,44 @@
...
+- Run docker-compose.yaml in package ***.docker***.
+- Run each service (check each service for more information).
+- Profit.
+
+## Technologies
+- SpringBoot
+- PostgreSQL
+- Docker
+- Apache Kafka
```
Comment (adsukharev) [inline README.md]: Let's make it in a different way and add some other technologies:
```
## Technologies

[Server] SpringBoot, Gradle

[Frontend] React, Typescript

[Databases] Postgres, MongoDB

[Tools] Docker, Kafka, API-Gateway, Circuit-breaker, Service Discovery
```

BTW, add gradle, maven and git to section `Tools` to your CV

File: README.md (hunk 5 above)
Comment (adsukharev) [inline README.md]:
```suggestion
- **Service Discovery** - allows services to find and communicate with each other
```

File: README.md [hunk 12]
```diff
@@ -1,2 +1,44 @@
...
+
+## Structure
+Project contains 7 services,
+each of which is responsible for performing key tasks to maintain the project's performance.
+
+Architecture: https://drive.google.com/file/d/1Jn5VrUUoQQtLY0N9Jk_7VIYwRXwG2yOq/view?usp=sharing
+- **User manager** - stores and processes user information.
+- **Payment manager** - stores and processes information about wallets with further sending of data to other services
+  (Notification and Transaction manager).
+- **Notification manager** - receives data from ***Apache Kafka*** (data from the Payment manager is first sent to Kafka),
```
Comment (adsukharev) [inline README.md]:
```suggestion
- **Notification manager** - send emails to the users
```

File: README.md [hunk 13]
```diff
@@ -1,2 +1,44 @@
...
+- [x] Create tests with code coverage for each service over 80%.
+- [ ] Implementation of authentication and authorization using JWT.
+- [ ] Implementation of API-Gateway and Service Discovery to simplify communication between services and clients.
+
+## Structure
+Project contains 7 services,
+each of which is responsible for performing key tasks to maintain the project's performance.
+
+Architecture: https://drive.google.com/file/d/1Jn5VrUUoQQtLY0N9Jk_7VIYwRXwG2yOq/view?usp=sharing
+- **User manager** - stores and processes user information.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- **User manager** - handles user information.
```

File: README.md [hunk 14]
```diff
@@ -1,2 +1,44 @@
...
+- **Payment manager** - stores and processes information about wallets with further sending of data to other services
```
Comment (adsukharev) [inline README.md]:
```suggestion
- **Payment manager** - performs payment process
```

File: README.md [hunk 15]
```diff
@@ -1,2 +1,44 @@
...
+  (Notification and Transaction manager).
+- **Notification manager** - receives data from ***Apache Kafka*** (data from the Payment manager is first sent to Kafka),
+  then sends a message to the user's email with information about the operations performed on the wallet.
+- **Transaction manager** - stores information about completed transactions over wallets.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- **Transaction manager** - stores information about payment transactions.
```

File: README.md [hunk 16]
```diff
@@ -1,2 +1,44 @@
...
+- **QR generator** - generates a QR-code based on transaction information: amount of money, sender, recipient.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- **QR generator** - generates a QR-code based on transaction information
```

File: README.md [hunk 17]
```diff
@@ -1,2 +1,44 @@
...
+- **Auth manager** - performs user authentication and authorization in the system.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- **Auth manager** - performs authentication and authorization
```

File: notification-manager/README.md [hunk 18]
```diff
@@ -1,7 +1,8 @@
 # Notification manager
-Service for sending emails about payments
+### Service for sending emails about payments.
```
Comment (adsukharev) [inline notification-manager/README.md]:
```suggestion
### Service for sending emails
```

not only payments, maybe registration and so on

File: notification-manager/README.md [hunk 19]
```diff
@@ -1,7 +1,8 @@
...
 
-### Start service
-- Open package resources
-- Write username (your email) and password in application.yaml
-- start service
\ No newline at end of file
+## How to run
+- Open file ***src/main/resources/application.yaml***.
+- Write ***username*** (your email) and ***password*** in application.yaml.
```
Comment (adsukharev) [inline notification-manager/README.md:8]: I assume you change your gmail profile a bit to allow 3rd party apps communicate with your email.
If yes, add on the next line that they have to setup their host email as well

**Review by adsukharev — COMMENTED** [review]
In general good, but I added some suggestions to make it not so wordy

File: README.md [hunk 20]
```diff
@@ -1,2 +1,44 @@
 # QPay
-### Payment system with QR code
+### Quick payment system by generating a QR code.
+
+## Description
+The project is aimed at implementing a payment system with QR code generation in a microservice architecture.
+
+## Goals
+- [x] Apply a monorepositional approach to the interaction between services.
```
Comment (adsukharev) [inline README.md]:
```suggestion
- [x] Apply a monorepo approach 
```
I don't think the word `monorepositional`  exist.

**Review by adsukharev — APPROVED** [review]

**Review by kostya4201337 — APPROVED** [review]

**Pull request merged**